### PR TITLE
TRAIN-268: Make exceptions return json instead of text.

### PR DESCRIPTION
**Pull request by r-barlow**

### CODE REVIEW

File: src/main/java/za/co/neslotech/spring/trainspringbootinsight/handler/RestExceptionHandler.java [hunk 1]
```diff
@@ -14,13 +16,19 @@
...
 
     @ExceptionHandler(EntityNotFoundException.class)
     protected ResponseEntity<Object> handleEntityNotFound(final EntityNotFoundException exception,
                                                           final WebRequest request) {
-        return handleExceptionInternal(exception, exception.getMessage(), new HttpHeaders(),
+
+        final var headers = new HttpHeaders();
+        headers.add("Content-Type", "application/json");
+
+        return handleExceptionInternal(exception, String.format("{ \"message\": \"%s\"}", exception.getMessage()), headers,
```
Comment (colsen-neslo) [inline src/main/java/za/co/neslotech/spring/trainspringbootinsight/handler/RestExceptionHandler.java]: So an alternative to creating a formatted string response json object, you can create a generic API error object that can consume some information like the error message, the reason for the error, the response type, etc. You can then just return a repsponse entity of the error type with all the information and it should then be mapped to json.

File: src/main/resources/application.properties [hunk 2]
```diff
@@ -7,7 +7,7 @@ jwt.secret=${JWT_SECRET}
 spring.jpa.properties.hibernate.dialect = org.hibernate.dialect.PostgreSQLDialect
 spring.jpa.properties.hibernate.show-sql=true
 spring.jpa.properties.hibernate.format_sql=true
-spring.jpa.hibernate.ddl-auto=create-drop
+spring.jpa.hibernate.ddl-auto=none
```
Comment (colsen-neslo) [inline src/main/resources/application.properties]: Should this not be set to update?

**Review by colsen-neslo — CHANGES_REQUESTED** [review]

File: src/main/java/za/co/neslotech/spring/trainspringbootinsight/handler/RestExceptionHandler.java [hunk 3]
```diff
@@ -6,45 +6,62 @@
...
 
     @ExceptionHandler(EntityNotFoundException.class)
     protected ResponseEntity<Object> handleEntityNotFound(final EntityNotFoundException exception,
                                                           final WebRequest request) {
-        return handleExceptionInternal(exception, exception.getMessage(), new HttpHeaders(),
-                                       HttpStatus.NOT_FOUND, request);
+
+        return handleExceptionInternal(exception,
+                                       new RestException(exception.getMessage(), HttpStatus.NOT_FOUND.value()),
+                                       new HttpHeaders(), HttpStatus.NOT_FOUND, request);
```
Comment (colsen-neslo) [inline src/main/java/za/co/neslotech/spring/trainspringbootinsight/handler/RestExceptionHandler.java:28]: What does this method do internally? Does it respond with the same status that you are providing?

Comment (r-barlow) [reply src/main/java/za/co/neslotech/spring/trainspringbootinsight/handler/RestExceptionHandler.java:28]: It returns a ResponseEntity<Object> with the status code, object and defined headers. It is a built in method.

**Review by JOlsen-Neslo — APPROVED** [review]

**Review by TristanCarswell — APPROVED** [review]

**Review by colsen-neslo — APPROVED** [review]

**Pull request merged**